R/utils.R

```diff
@@ -454,3 +454,50 @@ replace_or_append <- function(lines, pattern, replacement) {
   }
   lines
 }
+
+# Request an OAuth access token for the given resource from Posit Connect. The
+# OAuth token will belong to the user owning the given Shiny session.
+connect_viewer_token <- function(session, resource) {
+  # Ensure we're running on Connect.
+  server_url <- Sys.getenv("CONNECT_SERVER")
+  api_key <- Sys.getenv("CONNECT_API_KEY")
+  if (nchar(server_url) == 0 || nchar(api_key) == 0) {
```

Comment:
Or maybe there's some debug mode you can activate that additionally prints a message for each of these early returns?

Reply:
Modified this a bit so that there is only one early return, and it prints an informative message.